free-1.md

```diff
@@ -1,68 +1,198 @@
+   that programming languages so far have just been about users telling
+   computers what to do. Allowing for computers to give feedback to users
+   expands the language to allow for 2-way communication. And the way the
+   computer sends back information will need to be designed well to make it
+   most useful to the users.
 
 ### Why you?
-_What excites you about this idea? How did you come up with it?_
 
+I've always enjoyed geometry, and am excited about building an interactive
+system which makes computers a better platform for exploring mathematical ideas
+that are not fully formed yet.
 
 ### Domain
-_Describe the project's domain in five words._
 
+Interactively creating geometric drawings
 
 ### Interface (syntax)
-_How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
 
+Users could interact with the language statically or interactively.
+
+In either case they would write statements like:
+
+```
+circle x with center C
+line y tangent to C at point D
+line z through points C and D
+```
+
+I really like language idea because it mirrors the way that mathematicians
+write about geometry. Who hasn't seen something like:
+
+Let x be a circle centered at C. Let y be a line tangent to x at point D. Let z
+be a line through C and D. ...more proof...
 
 ### Operation (semantics)
-_What might happen when a program runs? How does a program interact with the
-user? What kinds of errors might occur, and how might they be communicated to
-the user?_
 
+So operation would work in two ways.
+
+The first of these is the static mode. People write stuff like above and
+trigger compilation which generates an image or LaTeX code, or something like
+that.
+
+The more interesting would be interactive mode, where as the users enter each
+line the image is drawn for them to see what they have, as well as to indicate
+potentially interesting thing to do next (IE, highlighting the tangent point
+after drawing a tangent). Using this information the users can issue special
+interactive commands that act on those possibilities.
+
+At the end of the session the user could export the image, LaTeX, or DSL code
+that would have created the image they ended up creating interactively.
+
+When errors occurs statically, they would be reported along with the part of
+the program which caused the error. When they occur interactively they would be
+reported right away, and the current drawing would remain unchanged.
 
 ### Expressiveness
-_What should be easy to do in this language? What should be possible, but
-difficult? What should be impossible or very difficult?_
 
+It should be easy (and should feel natural) to make simple geometric drawings
+and format them in simple ways (rotation, size, cropping, labelling, etc).
+
+It should be hard to make really complex drawings with lots of parts. Visual
+clutter is not something that is desirable in geometry.
```

Comment:
I'm not 100% sure what you're getting at here.

If you mean putting multiple drawings together, I'd say that shouldn't be part of the language. Geometric drawings are fundamentally all in one piece.

If you mean defining macros (perpendicular bisector, angle bisector), I think that would be great!